### ISSUE TITLE
ReferenceError: path is not defined

### ISSUE THREAD

**Issue by brunodantas** [BlueHatbRit/mdpdf#61]
Trying to run on Ubuntu 16:

```
ReferenceError: path is not defined
    at processSrc (/usr/local/lib/node_modules/mdpdf/src/utils.js:51:23)
    at Object.$.each (/usr/local/lib/node_modules/mdpdf/src/utils.js:59:23)
    at initialize.exports.each (/usr/local/lib/node_modules/mdpdf/node_modules/cheerio/lib/api/traversing.js:300:24)
    at Object.qualifyImgSources (/usr/local/lib/node_modules/mdpdf/src/utils.js:58:12)
    at prepareHeader.then.then.then.then.mdDoc (/usr/local/lib/node_modules/mdpdf/src/index.js:116:23)
    at tryCatcher (/usr/local/lib/node_modules/mdpdf/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:694:18)
    at Promise._fulfill (/usr/local/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:638:18)
    at /usr/local/lib/node_modules/mdpdf/node_modules/bluebird/js/release/nodeback.js:42:21
    at FSReqCallback.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:54:3)
```
with `node v11.8.0`, `npm 6.5.0`

**Comment by BlueHatbRit** [comment]
@brunodantas what did you try to run, I'm guessing this is from calling via cli rather than the js api?

**Comment by brunodantas** [comment]
Yes, I ran:

`$ mdpdf file.md`

**Comment by lehenric** [comment]
I am having same issue with `2.0.1`, reverting back to `1.7.3` helped.

**Comment by BlueHatbRit** [comment]
Thanks, I'll check it out now

**Comment by BlueHatbRit** [comment]
Hey @brunodantas and @lehenric could you give me the following details please?

1. Operating system
2. Nodejs version
3. Installation method (npm local install, npm global install, yarn etc)

I'm struggling to recreate this locally at the moment.

**Comment by pixelsonly** [comment]
@BlueHatbRit I'm also running into this issue. Here are my specific details.

`macOS 10.14.3`
`NodeJS v10.15.0`
`yarn v1.13.0`

Installed `mdpdf` globally with Yarn
```
$ mdpdf README.md --format=Letter
```

When running via `cli` is also see a similar error:
```
ReferenceError: path is not defined
    at processSrc (/Users/ryan/.config/yarn/global/node_modules/mdpdf/src/utils.js:51:23)
    at Object.$.each (/Users/ryan/.config/yarn/global/node_modules/mdpdf/src/utils.js:59:23)
    at initialize.exports.each (/Users/ryan/.config/yarn/global/node_modules/cheerio/lib/api/traversing.js:300:24)
    at Object.qualifyImgSources (/Users/ryan/.config/yarn/global/node_modules/mdpdf/src/utils.js:58:12)
    at prepareHeader.then.then.then.then.mdDoc (/Users/ryan/.config/yarn/global/node_modules/mdpdf/src/index.js:116:23)
    at tryCatcher (/Users/ryan/.config/yarn/global/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/ryan/.config/yarn/global/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/ryan/.config/yarn/global/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/ryan/.config/yarn/global/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/ryan/.config/yarn/global/node_modules/bluebird/js/release/promise.js:694:18)
    at Promise._fulfill (/Users/ryan/.config/yarn/global/node_modules/bluebird/js/release/promise.js:638:18)
    at /Users/ryan/.config/yarn/global/node_modules/bluebird/js/release/nodeback.js:42:21
    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)
```

**Comment by lehenric** [comment]
@BlueHatbRit 
1. Fedora 29 on `4.20.6-200.fc29.x86_64` kernel
2. node v10.11.0
    npm 6.4.1
3.  Installed via `npm` globally and running only through `cli`
```bash
npm install mdpdf -g
npm install mdpdf@1.7.3 -g
...
mdpdf README.md
```
Don't know if it's really important but I have `node` installed in my home directory. 

Also here is stacktrace from error:
```
ReferenceError: path is not defined
    at processSrc (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/src/utils.js:51:23)
    at Object.$.each (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/src/utils.js:59:23)
    at initialize.exports.each (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/cheerio/lib/api/traversing.js:300:24)
    at Object.qualifyImgSources (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/src/utils.js:58:12)
    at prepareHeader.then.then.then.then.mdDoc (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/src/index.js:116:23)
    at tryCatcher (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:694:18)
    at Promise._fulfill (/home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/bluebird/js/release/promise.js:638:18)
    at /home/lehenric/nodejs/node-v10.11.0-linux-x64/lib/node_modules/mdpdf/node_modules/bluebird/js/release/nodeback.js:42:21
    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)
```

**Comment by BlueHatbRit** [comment]
Thanks so much, I'll take a look and see if I can get this reproducing!

**Comment by pixelsonly** [comment]
@BlueHatbRit looking like the common denominator may be the global install :)

**Comment by BlueHatbRit** [comment]
@pixelsonly it does seem that but I've installed globally on several machines without an issue using npm which is confusing and makes this awkward so I'm going to take a look at the code as well and see what's going on.

Does anyone have an example markdown file they can provide that they've been using? I have a feeling it might have something to do with image processing.

**Comment by lehenric** [comment]
@BlueHatbRit 

>  I have a feeling it might have something to do with image processing.

That's quite right, I've just tried using `2.0.1` on other `md` file without images with success. 

Unfortunately I cannot share `md` file I was trying to convert but I included images with this syntax:
```md
![](png/image.png)
```
Maybe something with images in different path than `md` file?

**Comment by BlueHatbRit** [comment]
No worries that's more than enough for me to go on. Mdpdf has to convert paths from relative to aboslute when processing pdf's due to puppeteer so judging by the stack trace that's somehow returning `undefined` are you able to show me the directory tree and the exact (well, masked) path you're using in the markdown? @lehenric 

Thanks so much for your help debugging this everyone, it's made this a lot easier!

**Comment by lehenric** [comment]
Sure @BlueHatbRit,  (masked) output of `tree`:
```
.
├── csv
│   ├── data1.csv
│   ├── data2.csv
│   ├── data3.csv
│   └── data4.csv
├── plotGraphs.py
├── images
│   ├── image1.png
│   ├── image2.png
│   ├── image3.png
│   └── image4.png
└── report.md
```

And path is almost same as in my previous comment:
```
![](images/image1.png)
```

**Comment by BlueHatbRit** [comment]
Thanks @lehenric! For a quick fix, I think you could use `./images/image1.png` which I _think_ should fix it, I'll take a look and get a proper fix for this though.

**Comment by pixelsonly** [comment]
@BlueHatbRit I think you're onto something there! I failed to mention that my markdown source was also loading some relative images.

**Comment by BlueHatbRit** [comment]
If prepending the `./` worked then I think it will be this issue, did it work? 😄

**Comment by lehenric** [comment]
>  For a quick fix, I think you could use `./images/image1.png` which I _think_ should fix it,

Unfortunately this does not work and I get same error as without explicitly saying it's in current directory. Also stacktrace is same as previous one.

I use version `1.7.3` as a workaround right now.

**Comment by BlueHatbRit** [comment]
Sorry this has taken so long, I've found the issue and made the fix. I'll be making a new release in the next 12 hours.

**Comment by BlueHatbRit** [comment]
I've pushed a fix to the master branch which is usable if anyone wants to get access immedietely. You should be able to clone the repo, `npm install` and `npm link`. I'll make a release to npm in a few hours, I forgot to bring my signing key with me today.

**Comment by lehenric** [comment]
Thanks for a fix @BlueHatbRit! Tried out new version from master and it works.

Also when creating `pdf` I got some, I guess, debugging output. Output is something like this:
```js
./image/image1.png
{ ghStyle: true,
  defaultStyle: true,
  source:
   '/absolute/path/to/working/dir/report.md',
  destination:
   '/absolute/path/to/working/dir/report.pdf',
  styles: null,
  header: undefined,
  footer: undefined,
  noEmoji: false,
  debug: null,
  pdf:
   { format: 'A4',
     orientation: 'portrait',
     quality: '100',
     base: 'file:/home/lehenric/gits/mdpdf/bin/assets/',
     header: { height: null },
     footer: { height: null },
     border: { top: '20mm', left: '20mm', bottom: '20mm', right: '20mm' } },
  assetDir: '/absolute/path/to/working/dir/' }
```
I know it's not really a bug or anything but if I recall correctly, previous version with working image inclusion did not have this output.

Also installed via `npm install && npm link`. My knowledge about npm, it's packaging and installing via `npm install` is limited so don't really know if it's desired behavior.

**Comment by BlueHatbRit** [comment]
@lehenric I left this in for a moment for anyone that wanted to test, it'll get removed before the release is made, thanks for pointing it out though! I just wanted to be able to get a bit of debug information if anyone did try it out and had issues again.

**Comment by BlueHatbRit** [comment]
I've removed this logging now on master if you wanted to pull an update. You won't need to `npm install` or `npm link` again. A release will be made once I get home and have access to my signing key and then you can continue on as normal!

**Comment by BlueHatbRit** [comment]
This has been fixed in version 2.0.2 which is now published to npm. Thanks for all your help and contributions!
